src/sync/impl/apple/network_reachability_observer.cpp

```diff
@@ -48,10 +50,17 @@ NetworkReachabilityStatus reachability_status_for_flags(SCNetworkReachabilityFla
 
 } // (anonymous namespace)
 
-NetworkReachabilityObserver::NetworkReachabilityObserver(util::Optional<std::string> hostname,
-                                                         std::function<void (const NetworkReachabilityStatus)> handler)
+NetworkReachabilityObserver& NetworkReachabilityObserver::shared(util::Optional<util::Logger&> logger_ref)
```

Comment:
I figured we weren't using or planning to use the hostnames functionality. Maybe it makes more sense to not have singletons, though, and just have each instance of an observer associated with a single callback (like before).